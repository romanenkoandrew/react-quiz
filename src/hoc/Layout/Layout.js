import React, { Component } from "react";
import styles from "./Layout.module.css";
import MenuToggle from "../../components/Navigation/MenuToggle/MenuToggle";
import Drawer from "../../components/Navigation/Drawer/Drawer";
import { connect } from "react-redux";

class Layout extends Component {
  state = { menu: false };
  onToggle = () => {
    this.setState({ menu: !this.state.menu });
  };
  menuCloseHandler = () => {
    this.setState({ menu: false });
  };
  render() {
    return (
      <div className={styles.Layout}>
        <Drawer
          isOpen={this.state.menu}
          onClose={this.menuCloseHandler}
          isAuthenticated={this.props.isAuthenticated}
        />
        <MenuToggle onToggle={this.onToggle} isOpen={this.state.menu} />
        <main>{this.props.children}</main>
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return { isAuthenticated: !!state.auth.token };
};
export default connect(mapStateToProps)(Layout);
