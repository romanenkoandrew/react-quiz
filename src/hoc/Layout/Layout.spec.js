import React from "react";
import { shallow } from "enzyme";
import { Layout } from "./Layout";
import LayoutConnect from "./Layout";
import configureStore from "redux-mock-store";

const mockStore = configureStore();

describe("Layout component", () => {
  let component, instance;
  beforeEach(() => {
    component = shallow(<Layout children={<div>Test</div>} />);
    instance = component.instance();
  });
  it("should render Layout", () => {
    expect(component).toMatchSnapshot();
  });
  it("Layout method onToggle", () => {
    expect(instance.state.menu).toBe(false);
    instance.onToggle();
    expect(instance.state.menu).toBe(true);
  });
  it("Layout method menuCloseHandler", () => {
    instance.setState({ menu: true });
    expect(instance.state.menu).toBe(true);
    instance.menuCloseHandler();
    expect(instance.state.menu).toBe(false);
  });
  it("should isAuthenticated in Layout props", () => {
    const store = mockStore({
      isAuthenticated: true,
      auth: { token: "" },
    });
    const component2 = shallow(
      <LayoutConnect store={store} children={<div>Test</div>} />
    );
    expect(component2.children().props().isAuthenticated).toBe(false);
  });
});
