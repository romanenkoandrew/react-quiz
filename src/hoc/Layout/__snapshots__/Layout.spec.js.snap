// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render Layout 1`] = `
<div
  className="Layout"
>
  <Drawer
    isOpen={false}
    onClose={[Function]}
  />
  <MenuToggle
    isOpen={false}
    onToggle={[Function]}
  />
  <main>
    <div>
      Test
    </div>
  </main>
</div>
`;
